File: FastReportElement.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Xml;
using System.Xml.Linq;

public class FastReportElement
{
    private string XMLFile;
    private XmlDocument xml = null;
    private List<string> allTextObjects = new List<string>();
    public static string ReportPath { get; set; }

    public FastReportElement(string xmlFile)
    {
        this.XMLFile = xmlFile;
        this.LoadXML();
    }

    public static void SetTextObject(string objectName, string value)
    {
        XmlDocument xml = new XmlDocument();
        xml.Load(ReportPath);
        XmlElement elm = xml.DocumentElement;

        XDocument xmlDoc = XDocument.Parse(xml.InnerXml);

        var items = from item in xmlDoc.Descendants("TextObject")
                    where item.Attribute("Name").Value == objectName
                    select item;

        foreach (XElement itemElement in items)
            itemElement.SetAttributeValue("Text", value);

        xmlDoc.Save(ReportPath);
    }

    public static void SetCellObject(string objectName, string value)
    {
        XmlDocument xml = new XmlDocument();
        xml.Load(ReportPath);
        XmlElement elm = xml.DocumentElement;

        XDocument xmlDoc = XDocument.Parse(xml.InnerXml);

        var items = from item in xmlDoc.Descendants("TableCell")
                    where item.Attribute("Name").Value == objectName
                    select item;

        foreach (XElement itemElement in items)
            itemElement.SetAttributeValue("Text", value);

        xmlDoc.Save(ReportPath);
    }

    public static void SetBarcodeValue(string objectName, string value)
    {
        XmlDocument xml = new XmlDocument();
        xml.Load(ReportPath);
        XmlElement elm = xml.DocumentElement;

        XDocument xmlDoc = XDocument.Parse(xml.InnerXml);

        var items = from item in xmlDoc.Descendants("BarcodeObject")
                    where item.Attribute("Name").Value == objectName
                    select item;

        foreach (XElement itemElement in items)
            itemElement.SetAttributeValue("Text", value);

        xmlDoc.Save(ReportPath);
    }


    public static void SetPictureObject(string objectName, string base64Image)
    {
        XmlDocument xml = new XmlDocument();
        xml.Load(ReportPath);
        XmlElement elm = xml.DocumentElement;
        XDocument xmlDoc = XDocument.Parse(xml.InnerXml);

        var items = from item in xmlDoc.Descendants("PictureObject")
                    where item.Attribute("Name").Value == objectName
                    select item;

        foreach (XElement itemElement in items)
            itemElement.SetAttributeValue("Image", base64Image);

        xmlDoc.Save(ReportPath);
    }

    public List<string> AllTextObjects
    {
        get
        {
            return allTextObjects;
        }
    }
    public void LoadXML()
    {
        allTextObjects.Clear();
        xml = new XmlDocument();
        xml.Load(XMLFile);

        XmlElement elm = xml.DocumentElement;
        XmlNodeList xnList = xml.SelectNodes("/Report/ReportPage/OverlayBand");
        XmlNodeList name = xml.GetElementsByTagName("TextObject");
        XmlNodeList objReportPage = xml.GetElementsByTagName("DataBand");

        foreach (XmlNode xl in name)
        {
            if (xl.Attributes["Text"] == null)
                AddAttribute(xl, "Text", string.Empty);

            var _text = xl.Attributes["Text"].InnerText;

            allTextObjects.Add(xl.Attributes["Name"].InnerText);
        }

        elm = null;
        xnList = null;
        name = null;
        objReportPage = null;

        GC.Collect();
        GC.WaitForPendingFinalizers();

    }
    protected void AddAttribute(XmlNode node, string attribute, string value)
    {
        if (node.NodeType == XmlNodeType.Element)
        {
            node.Attributes.Append(node.OwnerDocument.CreateAttribute(attribute));
            node.Attributes[attribute].Value = value;
            foreach (XmlNode subnode in node.ChildNodes)
            {
                AddAttribute(subnode, attribute, value);
            }
        }
    }
}

File: FastReportElement.vb
'''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''
'FileName: FastReportElement.cs
'FileType: Visual C# Source file
'Author : Kirati Petkong
'Copy Rights : (c) 2021 Software & Scale Engineering Co.,Ltd
'Description : Fast Report Element object Redirect fix use for instead from problem offical solution.
'''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''

Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Xml
Imports System.Xml.Linq

Public Class FastReportElement
    Private XMLFile As String
    Private xml As XmlDocument = Nothing
    Private allTextObjectsField As List(Of String) = New List(Of String)()
    Public Shared Property ReportPath As String

    Public Sub New(ByVal xmlFile As String)
        Me.XMLFile = xmlFile
        LoadXML()
    End Sub

    Public Shared Sub SetTextObject(ByVal objectName As String, ByVal value As String)
        Dim xml As XmlDocument = New XmlDocument()
        xml.Load(ReportPath)
        Dim elm As XmlElement = xml.DocumentElement
        Dim xmlDoc As XDocument = XDocument.Parse(xml.InnerXml)
        Dim items = From item In xmlDoc.Descendants("TextObject") Where Equals(item.Attribute("Name").Value, objectName) Select item

        For Each itemElement As XElement In items
            itemElement.SetAttributeValue("Text", value)
        Next

        xmlDoc.Save(ReportPath)
    End Sub

    Public Shared Sub SetCellObject(ByVal objectName As String, ByVal value As String)
        Dim xml As XmlDocument = New XmlDocument()
        xml.Load(ReportPath)
        Dim elm As XmlElement = xml.DocumentElement
        Dim xmlDoc As XDocument = XDocument.Parse(xml.InnerXml)
        Dim items = From item In xmlDoc.Descendants("TableCell") Where Equals(item.Attribute("Name").Value, objectName) Select item

        For Each itemElement As XElement In items
            itemElement.SetAttributeValue("Text", value)
        Next

        xmlDoc.Save(ReportPath)
    End Sub

    Public Shared Sub SetBarcodeValue(ByVal objectName As String, ByVal value As String)
        Dim xml As XmlDocument = New XmlDocument()
        xml.Load(ReportPath)
        Dim elm As XmlElement = xml.DocumentElement
        Dim xmlDoc As XDocument = XDocument.Parse(xml.InnerXml)
        Dim items = From item In xmlDoc.Descendants("BarcodeObject") Where Equals(item.Attribute("Name").Value, objectName) Select item

        For Each itemElement As XElement In items
            itemElement.SetAttributeValue("Text", value)
        Next

        xmlDoc.Save(ReportPath)
    End Sub

    Public Shared Sub SetPictureObject(ByVal objectName As String, ByVal base64Image As String)
        Dim xml As XmlDocument = New XmlDocument()
        xml.Load(ReportPath)
        Dim elm As XmlElement = xml.DocumentElement
        Dim xmlDoc As XDocument = XDocument.Parse(xml.InnerXml)
        Dim items = From item In xmlDoc.Descendants("PictureObject") Where Equals(item.Attribute("Name").Value, objectName) Select item

        For Each itemElement As XElement In items
            itemElement.SetAttributeValue("Image", base64Image)
        Next

        xmlDoc.Save(ReportPath)
    End Sub

    Public ReadOnly Property AllTextObjects As List(Of String)
        Get
            Return allTextObjectsField
        End Get
    End Property

    Public Sub LoadXML()
        allTextObjectsField.Clear()
        xml = New XmlDocument()
        xml.Load(XMLFile)
        Dim elm As XmlElement = xml.DocumentElement
        Dim xnList As XmlNodeList = xml.SelectNodes("/Report/ReportPage/OverlayBand")
        Dim name As XmlNodeList = xml.GetElementsByTagName("TextObject")
        Dim objReportPage As XmlNodeList = xml.GetElementsByTagName("DataBand")

        For Each xl As XmlNode In name
            If xl.Attributes("Text") Is Nothing Then AddAttribute(xl, "Text", String.Empty)
            Dim _text = xl.Attributes("Text").InnerText
            allTextObjectsField.Add(xl.Attributes("Name").InnerText)
        Next

        elm = Nothing
        xnList = Nothing
        name = Nothing
        objReportPage = Nothing
        GC.Collect()
        GC.WaitForPendingFinalizers()
    End Sub

    Protected Sub AddAttribute(ByVal node As XmlNode, ByVal attribute As String, ByVal value As String)
        If node.NodeType = XmlNodeType.Element Then
            node.Attributes.Append(node.OwnerDocument.CreateAttribute(attribute))
            node.Attributes(attribute).Value = value

            For Each subnode As XmlNode In node.ChildNodes
                AddAttribute(subnode, attribute, value)
            Next
        End If
    End Sub
End Class
